### PR TITLE
Blog App - Controllers

### PR DESCRIPTION
**Pull request by Izebiafe**
In this PR ;
I created a post controller and a user controller

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class PostsController < ApplicationController
```
Comment (DammyShittu) [inline app/controllers/posts_controller.rb:1]: - [ ] Your controller actions cannot exist in a vacuum. They need to be linked to routes. In your routes file.  Kindly define your routes.

  ```
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
  ```
  
  image

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  before_action :find_user, only: [:show]
+
+  def index
+    @users = User.all
+  end
+
+  def show; end
+
+  def new; end
+
```
Comment (DammyShittu) [inline app/controllers/users_controller.rb:11]: -  [ ] Kindly remove this action. We should only have index and show actions.

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  def find_user
+    @user = User.find(params[:id])
+  rescue ActiveRecord::RecordNotFound
+    # Handle the situation when the post is not found
+    flash[:alert] = 'User not found, back to users page'
+    redirect_to users_path
+  end
```
Comment (DammyShittu) [inline app/controllers/users_controller.rb:18]: - [ ] Kindly make this a private method as there is no route meant for it or remove it.

File: app/controllers/posts_controller.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    @user = User.find(params[:user_id])
+  end
+
+  def find_post
+    @post = @user.posts.find(params[:id])
+  rescue ActiveRecord::RecordNotFound
+    # Handle the situation when the post is not found
+    flash[:alert] = 'Post not found, back to posts page'
+    redirect_to user_posts_path(@user)
+  end
```
Comment (DammyShittu) [inline app/controllers/posts_controller.rb:23]: - [ ] Kindly make these private methods as there are no routes meant for them

File: app/controllers/posts_controller.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  before_action :find_user, only: %i[index show]
+  before_action :find_post, only: [:show]
+
+  def index
+    @posts = @user.posts
+  end
+
+  def show; end
+
+  def new; end
```
Comment (DammyShittu) [inline app/controllers/posts_controller.rb:11]: - [ ] Kindly remove this `new` method here. We should only have index and show actions.

File: app/controllers/posts_controller.rb [hunk 6]
```diff
@@ -0,0 +1,24 @@
+class PostsController < ApplicationController
+  before_action :find_user, only: %i[index show]
+  before_action :find_post, only: [:show]
+
+  def index
+    @posts = @user.posts
+  end
+
+  def show; end
```
Comment (DammyShittu) [inline app/controllers/posts_controller.rb:9]: - [ ] Kindly create view pages for these two actions.

File: app/controllers/users_controller.rb [hunk 7]
```diff
@@ -0,0 +1,19 @@
+class UsersController < ApplicationController
+  before_action :find_user, only: [:show]
+
+  def index
+    @users = User.all
+  end
+
+  def show; end
```
Comment (DammyShittu) [inline app/controllers/users_controller.rb:8]: - [ ] Kindly create view pages for these two actions.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @Izebiafe,

## Highlights
✔️ Controller methods

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Davidosky007 — APPROVED** [review]
Hi @Izebiafe 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* All controllers created properly 👍 
* No linter errors 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**